smif/sample_project/models/energy_demand.py

```diff
@@ -11,27 +12,62 @@ def initialise(self, initial_conditions):
         pass
 
     def simulate(self, data):
-        # receive population, energy_demand at annual/national
-        self.logger.info("EDMWrapper received inputs in %s", data.current_timestep)
+
+        # Get the current timestep
+
+        now = data.current_timestep
+        self.logger.info("EDMWrapper received inputs in %s",
+                         now)
+
+        # Demonstrates how to get the value for a model parameter
+        parameter_value = data.get_parameter('smart_meter_savings')
+        self.logger.info('Smart meter savings: %s', parameter_value)
+
+        # Demonstrates how to get the value for a model input
+        # (defaults to the current time period)
+        current_energy_demand = data.get_data('energy_demand')
+        self.logger.info("Current energy demand in %s is %s",
+                         now, current_energy_demand)
+
+        # Demonstrates how to get the value for a model input from the base
+        # timeperiod
+        base_energy_demand = data.get_data('energy_demand',
+                                           RelativeTimestep.BASE)
+        base_year = RelativeTimestep.BASE.resolve_relative_to(now,
+                                                              data.timesteps)
+        self.logger.info("Base year energy demand in %s was %s", base_year,
+                         base_energy_demand)
+
+        # Demonstrates how to get the value for a model input from the previous
+        # timeperiod
+        if now > base_year:
+            prev_energy_demand = data.get_data('energy_demand',
+                                               RelativeTimestep.PREVIOUS)
+            prev_year = RelativeTimestep.PREVIOUS.resolve_relative_to(
```

Comment:
Getting the value of a previous year or the base year could be streamlined - perhaps adding a helper method to the data_handle?

Reply:
@willu47 how do the helper properties/methods below look?